File: Python3/프로그래머스/2/42577. 전화번호 목록/전화번호 목록.py
def solution(phone_book):
    phone_book.sort()
    # print("Sorted Phone Book:", phone_book)
    
    before = "None"
    for p in phone_book:
        if len(p) >= len(before) and before == p[:len(before)]:
            return False
        before = p
    
    
    return True

File: C++/프로그래머스/2/42577. 전화번호 목록/전화번호 목록.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

bool solution(vector<string> phone_book) {
    sort(phone_book.begin(), phone_book.end());
    bool flag;
    
    for(int i = 1; i < phone_book.size(); i++) {
        flag = false;
        
        for(int j = 0; j < phone_book[i - 1].size(); j++) { //이전 값과 비교
            if(phone_book[i].size() <= j || phone_book[i - 1][j] != phone_book[i][j]) {
                flag = true;
                break;
            }
        }
        if(!flag) {
            return false;
        }
    }
    
    return true;
}